napari/layers/vectors/vectors.py

```diff
@@ -701,7 +699,9 @@ def _update_thumbnail(self):
                     0, self._view_data.shape[0], self._max_vectors_thumbnail
                 )
                 vectors = copy(self._view_data[thumbnail_indices, :, -2:])
-                thumbnail_color_indices = self._view_indices[thumbnail_indices]
+                thumbnail_color_indices = np.array(self._view_indices)[
+                    thumbnail_indices
+                ]
```

Comment:
If you accept my initialization suggestion, then you can revert this back to the simpler:

```suggestion
                thumbnail_color_indices = self._view_indices[thumbnail_indices]
```